core/transaction_pool.go

```diff
@@ -71,7 +71,12 @@ type TxPool struct {
 	queue   map[common.Address]map[common.Hash]*types.Transaction
 }
 
-func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) *TxPool {
+func NewTxPool(eventMux *event.TypeMux, currentStateFn stateFn, gasLimitFn func() *big.Int) (*TxPool, error) {
+	currentState, err := currentStateFn()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Yeah, you're kind of right. Maybe txpool should keep a reference to BlockChain then (tight coupling FTW).

But this was the error that raised our attention, right? The state wasn't there when executing block tests.